### PR TITLE
feat: port in use

### PR DESCRIPTION
**Pull request by wconrad265**
🎉 Thanks for submitting a pull request! 🎉

#### Summary

This pull request addresses these issues found below

- https://github.com/netlify/cli/issues/6840
- https://github.com/netlify/cli/issues/6805

We (@tlane25) outline the problem in the comments in https://github.com/netlify/cli/issues/6840.

This pull request adds better error handling for users who open a port for a serverless function. 

---

For us to review and ship your PR efficiently, please perform the following steps:

- [ ] Open a [bug/issue](https://github.com/netlify/cli/issues/new/choose) before writing your code 🧑‍💻. This ensures we can discuss the changes and get feedback from everyone that should be involved. If you\`re fixing a typo or something that\`s on fire 🔥 (e.g. incident related), you can skip this step.
- [ ] Read the [contribution guidelines](../CONTRIBUTING.md) 📖. This ensures your code follows our style guide and
      passes our tests.
- [ ] Update or add tests (if any source code was changed or added) 🧪
- [ ] Update or add documentation (if features were changed or added) 📝
- [ ] Make sure the status checks below are successful ✅

**A picture of a cute animal (not mandatory, but encouraged)**

### CODE REVIEW

File: src/commands/main.ts [hunk 1]
```diff
@@ -40,27 +40,39 @@ import { createWatchCommand } from './watch/index.js'
 
 const SUGGESTION_TIMEOUT = 1e4
 
-process.on('uncaughtException', async (err) => {
+process.on('uncaughtException', async (err: any) => {
```
Comment (DanielSLew) [inline src/commands/main.ts:43]: Let's use a proper type here rather than adding in an `any`

**Review by DanielSLew — CHANGES_REQUESTED** [review]
Looks good, just a small nit about `any` types.

**Comment by wconrad265** [comment]
This has been updated with a new type.

File: layout.html [hunk 2]
```diff
@@ -1,4 +1,4 @@
-<!DOCTYPE html>
```
Comment (DanielSLew) [inline layout.html:1]: why was this changed?

File: src/commands/types.d.ts [hunk 3]
```diff
@@ -73,3 +73,11 @@ export type NetlifyOptions = {
   state: StateConfig
   frameworksAPIPaths: FrameworksAPIPaths
 }
+
+export interface AddressInUseError extends Error {
```
Comment (DanielSLew) [inline src/commands/types.d.ts:77]: I would just add this into the same error type as optional fields

File: package.json [hunk 4]
```diff
@@ -161,6 +161,8 @@
     "parse-github-url": "1.0.3",
     "parse-gitignore": "2.0.0",
     "path-key": "4.0.0",
+    "pnpm": "^9.12.2",
+    "prettier": "^3.3.3",
```
Comment (DanielSLew) [inline package.json:165]: why were these added for this PR?

File: src/commands/types.d.ts (hunk 3 above)
Comment (tlane25) [reply src/commands/types.d.ts:77]: The `Error` type lives in `~external/38d21a27-4378-4895-a56a-d0fa5726a379/lib.es2022.error.d.ts`. I think it's from an dependency/ external source.

File: layout.html (hunk 2 above)
Comment (wconrad265) [reply layout.html:1]: image

When we were trying to push this commit, we kept having a prettier format error for this file. This is what it recommended that we do, to update to a lowercase `doctype`. From my understanding, there is no functional difference between the two.

File: package.json (hunk 4 above)
Comment (wconrad265) [reply package.json]: I believe we were working on an new machine, and they were accidently not install locally, they have been removed in the update.

**Review by DanielSLew — APPROVED** [review]

**Comment by wconrad265** [comment]
The tests have been fixed. Ready for re-review

**Comment by wconrad265** [comment]
Look like this might be a flakey test. It passed in every single other environment, and passes the test locally. The tests were passing until I merged the latest version of main.

**Review by DanielSLew — APPROVED** [review]
